src/hugr/validate.rs

```diff
@@ -392,7 +392,13 @@ impl<'a> ValidationContext<'a> {
 
         let postorder = DfsPostOrder::new(&region, entry_node);
         let nodes_visited = postorder.iter(&region).filter(|n| *n != parent).count();
-        if nodes_visited != self.hugr.hierarchy.child_count(parent.index) {
+        // ModuleOp's (local Consts and Defs) should not be reachable from the Input node, so discount them
+        let non_module_op_count = self
+            .hugr
+            .children(parent)
+            .filter(|n| !OpTag::ModuleOp.contains(self.hugr.get_optype(*n).tag()))
```

Comment:
it seems somewhat incidental from the spec that `ModuleOp`s are the ones we wish to ignore, are we happy that it makes sense?

Also note if we go with this the `ConstInput` tag introduced in https://github.com/CQCL-DEV/hugr/blob/93c677c51c43ab37a09c8a118935b1d7566bb723/src/hugr/validate.rs#L342 can be removed.

Reply:
Well, I was thinking that all the other ModuleOp's *could* be "promoted" to be allowed locally - and it would make sense to, but we might choose not to because it makes life much harder for the linker - so this was a bit of future-proofing. Ick, indeed.

So I've removed ConstInput, and added a "ScopedDefn" category that I think says what we want quite neatly. How's thath?